plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+            try:
+                affinity_id_decider = int(affinity_id_decider)
+                if not (0 <= affinity_id_decider <= 2147483647):
+                    self.msg = (
+                        "The 'affinity_id_decider' should be from 0 to 2147483647 for the "
+                        "device '{device_ip}'".format(device_ip=device_ip)
+                    )
+                    self.status = "failed"
+                    return self.check_return_status()
+
+            except ValueError:
+                self.msg = (
+                    "The 'affinity_id_decider' should contain only digits 0-9 for the device '{ip}'."
+                    .format(ip=device_ip)
+                )
+                self.status = "failed"
+                return self.check_return_status()
+
+        if (not affinity_id_prime) ^ (not affinity_id_decider):
+            self.msg = (
+                "Either 'affinity_id_prime' or 'affinity_id_decider' is not accepted. If you wish "
+                "to add the affinity, pass both the 'affinity_id_prime' or 'affinity_id_decider' "
+                "for the device with IP '{ip}".format(ip=device_ip)
+            )
+            self.status = "failed"
+            return self.check_return_status()
+
+        sda_l3_handoff_info.update({
+            "interfaceName": transit_id,
+            "affinityIdPrime": affinity_id_prime,
+            "affinityIdDecider": affinity_id_decider,
+            "connectedToInternet": connected_to_internet,
+            "isMulticastOverTransitEnabled": is_multicast_over_transit_enabled,
+        })
+
+        return sda_l3_handoff_info
+
+    def get_ip_l3_handoff_params(self, fabric_id, network_id, device_details, device_config_index, fabric_name):
+        """
+        Get the L3 IP Handoff of the SDA fabric devices information from playbook.
+        Set the status and the msg before returning from the API
+        Check the return value of the API with check_return_status()
+
+        Parameters:
+            fabric_id (str): The Id of the fabric site.
+            network_id (str): The Id of the network device.
+            device_details (dict): Playbook details containing fabric devices details along
+            with the Border Settings, L2 Handoff, L3 SDA Handoff, L3 IP Handoff information.
+            device_config_index (int): The index of the device details in the device_config.
+            fabric_name (str): The name of the fabric site.
+        Returns:
+            ip_l3_handoff_info (list): The processed L3 Handoff with IP transit details from the user playbook.
+        Description:
+
+        """
+
+        ip_l3_handoff_info = []
+        borders_settings = device_details.get("borders_settings")
+        if not borders_settings:
+            return ip_l3_handoff_info
+
+        layer3_handoff_ip_transit = borders_settings.get("layer3_handoff_ip_transit")
+        if not layer3_handoff_ip_transit:
```

Comment:
self.log("No Layer 3 Handoff IP Transit found for device with IP: %s", device_details.get("device_ip"))